vnext/Microsoft.ReactNative/ReactCoreInjection.idl

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import "IReactContext.idl";
+import "ReactNativeHost.idl";
+import "IJSValueReader.idl";
+
+#include "DocString.h"
+
+namespace Microsoft.ReactNative
+{
+  [experimental]
+  [webhosthidden]
+  DOC_STRING(
+    "Used to inject platform specific implementations to create react-native targets targeting non-XAML platforms.")
+  interface IReactCoreInjection
+  {
+    DOC_STRING("Called when the UI batch is completed")
+    void OnBatchComplete(IReactPropertyBag properties);
+  }
+
+[experimental]
+[default_interface]
+[webhosthidden]
+DOC_STRING("Settings per each IReactViewHost associated with an IReactHost instance.")
+  runtimeclass ReactViewOptions {
+    DOC_STRING("Creates a new instance of @ReactViewOptions")
+    ReactViewOptions();
+
+    DOC_STRING("Name of a component registered in JavaScript via AppRegistry.registerComponent('ModuleName', () => ModuleName);")
+    String ComponentName;
+
+    DOC_STRING("Set of initial component properties. It is a JSON string.")
+      JSValueArgWriter InitialProps {
+      get; set;
+    };
+  };
+
+  [experimental]
+  [webhosthidden]
+  DOC_STRING(
+    "Used to implement a non-XAML platform ReactRootView.")
+    interface IReactViewInstance
+  {
+    DOC_STRING("Initialize ReactRootView with the reactInstance and view-specific settings")
+      void InitRootView(IReactContext context, ReactViewOptions viewOptions);
+
+    DOC_STRING("Update ReactRootView with changes in ReactInstance")
+      void UpdateRootView();
+
+    DOC_STRING("Uninitialize ReactRootView and destroy UI tree")
+      void UninitRootView();
+  }
+
```

Comment:
these methods don't deal with xaml specifics, so could this be made the same for both office and uwp?

Reply:
in other words, can we move the uwp implementation to the new API and have both implementations on equal footing?
seems like in the office case you'd do:
```
something.OnBatchComplete(instance->context.Properties)
```
whereas in the uwp case we do:
```
M::RN::GetNativeUIManager(instance->context).OnBatchComplete()
```

Seems close enough that we could simplify this to not have special cases :)

Reply:
Totally agreed.  I'm planning on backporting this change to 0.64, so I wanted to keep the initial change with as little code churn as possible.  Then build on top of it to clean up this kind of thing.